src/Catel.Core/Catel.Core.Shared/WeakReferences/WeakFunc.cs

```diff
@@ -88,7 +89,18 @@ public bool Execute(out TResult result)
             {
                 if (IsTargetAlive)
                 {
-                    result = (TResult)_action.DynamicInvoke(Target);
+                    try
+                    {
+                        result = (TResult)_action.DynamicInvoke(Target);
+                    }
+                    catch (TargetInvocationException e)
+                    {
+                        if (e.InnerException != null)
+                        {
+                            throw e.InnerException;
```

Comment:
This will not contain the whole stacktrace, but I believe we have no other option.